api/objects.proto

```diff
@@ -48,6 +48,13 @@ message Node {
 
 	// Certificate is the TLS certificate issued for the node, if any.
 	Certificate certificate = 8 [(gogoproto.nullable) = false];
+
+	// Role is the *observed* role for this node. It differs from the
+	// desired role set in Node.Spec.Role because the role here is only
+	// updated after the Raft member list has been reconciled with the
+	// desired role from the spec. Note that this doesn't show whether the
```

Comment:
Maybe add that if the role is being used to tell whether or not an action may be performed, the certificate should be verified. This field is mostly informational.